agent/engine/docker_image_manager.go

```diff
@@ -335,18 +350,31 @@ func (imageManager *dockerImageManager) deleteImage(imageID string, imageState *
 		seelog.Errorf("Image ID to be deleted is null")
 		return
 	}
-	seelog.Infof("Removing Image: %s", imageID)
-	err := imageManager.client.RemoveImage(imageID, removeImageTimeout)
-	if err != nil {
-		if err.Error() == imageNotFoundForDeletionError {
-			seelog.Errorf("Image already removed from the instance: %v", err)
-		} else {
-			seelog.Errorf("Error removing Image %v - %v", imageID, err)
-			delete(imageManager.imageStatesConsideredForDeletion, imageState.Image.ImageID)
-			return
+
+	whitelisted := false
+	for _, m := range imageManager.persistentImages {
+		if imageID == m {
+			seelog.Infof("Image was whitelisted for deletion, image: %s", imageID)
```

Comment:
"whitelisted for deletion" sounds like it will be deleted.  Maybe "Image %s marked to be preserved, skipping deletion"